### ISSUE TITLE
org.opensearch.test.rest.ClientYamlTestSuiteIT > test {p0=search/30_limits/Regexp length limit} flaky

### ISSUE THREAD

**Issue by DarshitChanpura** [opensearch-project/OpenSearch#5197]
Gradle check failed when executing `:rest-api-spec:yamlRestTest` task with following error:

```
[2022-11-10T10:54:34,265][INFO ][o.o.t.r.ClientYamlTestSuiteIT] [test] Stash dump on test failure [{
     "stash" : {
       "body" : {
         "error" : {
           "root_cause" : [
             {
               "type" : "too_long_http_line_exception",
               "reason" : "An HTTP line is larger than 4096 bytes."
             }
           ],
           "type" : "too_long_http_line_exception",
           "reason" : "An HTTP line is larger than 4096 bytes."
         },
         "status" : 400
       }
     }
  }]
.
.
.
org.opensearch.test.rest.ClientYamlTestSuiteIT > test {p0=search/30_limits/Regexp length limit} FAILED
    java.lang.AssertionError: Failure at [search/30_limits:121]: the error message was expected to match the provided regex but didn't
    Expected: The length of regex \[1110\]
         but: was "{root_cause=[{type=too_long_http_line_exception, reason=An HTTP line is larger than 4096 bytes.}], type=too_long_http_line_exception, reason=An HTTP line is larger than 4096 bytes.}"
        at __randomizedtesting.SeedInfo.seed([199143B1F7094BF6:91C57C6B59F5260E]:0)
        at org.opensearch.test.rest.yaml.OpenSearchClientYamlSuiteTestCase.executeSection(OpenSearchClientYamlSuiteTestCase.java:459)
        at org.opensearch.test.rest.yaml.OpenSearchClientYamlSuiteTestCase.test(OpenSearchClientYamlSuiteTestCase.java:432)
        at java.****/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
```

**To Reproduce**

1. Checkout `main`
2. Run `./gradlew ':rest-api-spec:yamlRestTest' --tests "org.opensearch.test.rest.ClientYamlTestSuiteIT.test {p0=search/30_limits/Regexp length limit}" -Dtests.seed=199143B1F7094BF6 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=sv -Dtests.timezone=America/Havana -Druntime.java=19`
3. Try with `-Druntime.java=17` and `-Druntime.java=14`, it should give same result same result

Refer this build for full log: https://build.ci.opensearch.org/job/gradle-check/6714/consoleFull

**Expected behavior**
Tests should not fail.

**Host/Environment (please complete the following information):**
Using latest commit https://github.com/opensearch-project/OpenSearch/commit/e10c9253d621af08698d5b0966de4599ec342318 on main

**Additional context**
Possibly related: https://github.com/elastic/elasticsearch/issues/1174
Exception class: https://netty.io/4.1/api/io/netty/handler/codec/http/TooLongHttpLineException.html

**Comment by dblock** [comment]
https://build.ci.opensearch.org/job/gradle-check/6767/
  https://build.ci.opensearch.org/job/gradle-check/6761/
  https://build.ci.opensearch.org/job/gradle-check/6740/

**Comment by adnapibar** [comment]
Fixed by #5278
